vnext/local-cli/runWindows/utils/autolink.js

```diff
@@ -6,95 +6,359 @@
+      changesNecessary;
+
+    // Generating project entries for solution
+    let projectsForSolution = [];
+
+    for (const dependencyName in windowsDependencies) {
+      // Process projects
+      windowsDependencies[dependencyName].projects.forEach(project => {
+        const dependencyProjectFile = path.join(
+          windowsDependencies[dependencyName].folder,
+          windowsDependencies[dependencyName].sourceDir,
+          project.projectFile,
+        );
+
+        projectsForSolution.push({
+          projectFile: dependencyProjectFile,
+          projectName: project.projectName,
+          projectLang: project.projectLang,
+          projectGuid: project.projectGuid,
+        });
+      });
+
+      // Process additional projects
+      windowsDependencies[dependencyName].additionalProjects.forEach(
+        project => {
+          const dependencyProjectFile = path.join(
+            windowsDependencies[dependencyName].folder,
+            windowsDependencies[dependencyName].sourceDir,
+            project.projectFile,
+          );
+
+          projectsForSolution.push({
+            projectFile: dependencyProjectFile,
+            projectName: project.projectName,
+            projectLang: project.projectLang,
+            projectGuid: project.projectGuid,
+          });
+        },
+      );
+    }
+
+    verboseMessage(
+      `Calculating ${chalk.bold(path.basename(solutionFile))} changes...`,
+      verbose,
+    );
+
+    projectsForSolution.forEach(project => {
+      const contentsChanged = vstools.addProjectToSolution(
+        solutionFile,
+        project,
+        verbose,
+        checkMode,
+      );
+      changesNecessary = changesNecessary || contentsChanged;
+    });
+
+    spinner.succeed();
+
+    if (!changesNecessary) {
+      console.log(
+        `${chalk.green('Success:')} No auto-linking changes necessary.`,
+      );
+    } else if (checkMode) {
+      console.log(
+        `${chalk.yellow(
+          'Warning:',
+        )} Auto-linking changes were necessary but ${chalk.bold(
+          '--check',
+        )} specified. Run ${chalk.bold(
+          "'npx react-native autolink-windows'",
+        )} to apply the changes.`,
+      );
+      ExitProcessWithStatusCode(0, verbose);
+    } else {
+      console.log(`${chalk.green('Success:')} Auto-linking changes completed.`);
+    }
   } catch (e) {
-    console.error('Parsing react-native config failed!');
-    console.error(e);
-    return;
+    spinner.fail();
+    console.log(chalk.red('Error:') + ' ' + e.toString());
+    ExitProcessWithStatusCode(1, verbose);
   }
+
+  return;
 }
 
 module.exports = {
-  updateAutoLink,
+  name: 'autolink-windows',
+  description: 'performs autolinking',
+  func: updateAutoLink,
+  options: [
+    {
+      command: '--logging',
+      description: 'Verbose output logging',
+      default: false,
+    },
+    {
+      command: '--check',
+      description: 'Check (only) if any autolinked files need to change',
```

Comment:
consider writing this as "Only check whether any autolinked files need to change", or "Check if any autolinked files need to change and exit"

Reply:
Done.